packages/objection/src/interpreters.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  Condition,
+  CompoundCondition,
+  FieldCondition,
+  Comparable
+} from '@ucast/core';
+import { Query, ObjectionOperator } from './interpreter';
+import { renameFields } from './utils';
+
+export const $eq: ObjectionOperator<FieldCondition> = (condition, query) => {
+  return query.where(condition.field, '=', condition.value);
+};
+
+export const $ne: ObjectionOperator<FieldCondition> = (condition, query) => {
+  return query.where(condition.field, 'Not', condition.value);
+};
+
+export const $lt: ObjectionOperator<FieldCondition<Comparable>> = (condition, query) => {
+  return query.where(condition.field, '<', condition.value);
+};
+
+export const $lte: ObjectionOperator<FieldCondition<Comparable>> = (condition, query) => {
+  return query.where(condition.field, '<=', condition.value);
+};
+
+export const $gt: ObjectionOperator<FieldCondition<Comparable>> = (condition, query) => {
+  return query.where(condition.field, '>', condition.value);
+};
+
+export const $gte: ObjectionOperator<FieldCondition<Comparable>> = (condition, query) => {
+  return query.where(condition.field, '>=', condition.value);
+};
+
+export const $exists: ObjectionOperator<FieldCondition<Comparable>> = (condition, query) => {
+  return query.where(condition.field, condition.value ? 'NotNull' : 'Null', condition.value);
+};
+
+export const $in: ObjectionOperator<FieldCondition<Comparable[]>> = (condition, query) => {
+  return query.where(condition.field, 'In', condition.value);
+};
+
+export const $nin: ObjectionOperator<FieldCondition<Comparable[]>> = (condition, query) => {
+  return query.where(condition.field, 'NotIn', condition.value);
+};
+
+export const $not: ObjectionOperator<CompoundCondition> = (node, query, { interpret }) => {
+  const notQuery = query.buildUsing('whereNot');
+  node.value.forEach(condition => interpret(condition, notQuery));
+  return query;
+};
+
+export const $and: ObjectionOperator<CompoundCondition> = (node, query, { interpret }) => {
+  return node.value.reduce((builder, condition) => interpret(condition, builder), query);
+};
+
+export const $or: ObjectionOperator<CompoundCondition> = (node, query, { interpret }) => {
+  const orQuery = query.buildUsing('orWhere', query.query.clone());
+  node.value.forEach(condition => interpret(condition, orQuery));
+  query.query.where(builder => orQuery.applyTo(builder));
+  return query;
+};
+
+export const $nor: ObjectionOperator<CompoundCondition> = (node, query, context) => {
+  query.query.whereNot(builder => $and(node, new Query(builder), context).query);
```

Comment:
should use `$or` not `$and`. Probably you will need to change it to something similar to `$or`. Otherwise we will have the same issue with joins

Reply:
Correct me if I'm wrong but logical nor is true when both operands are false, hence the $and.

```ts
const condition = new CompoundCondition('$nor', [
  new Field('$eq', 'age', 1),
  new Field('$eq', 'active', true)
])
// select * from table where not (age = 1 and active = true);
```

It matches the js interpreter
```ts
expect(interpret(condition, { age: 1, active: true })).to.be.false
expect(interpret(condition, { age: 1, active: false })).to.be.false
expect(interpret(condition, { age: 10, active: true })).to.be.false
expect(interpret(condition, { age: 2, active: false })).to.be.true
```

Reply:
nor is inverted OR. In your case it's inverted AND which is a bit different. Let's look at the table

| a  | b | OR    | AND | NOT OR | NOT AND |
| -- | -- | ------- | ------ | ------------ | -------------- |
| 1 | 1  | 1       | 1      | 0              | 0               |   
| 1 | 0 | 1        |  0     | 0              | 1              |
| 0 | 1 | 1        |  0     | 0              | 1              |
| 0 | 0 | 0        |  0     | 1              | 1              |

Reply:
You're right, my bad. Fixed.